source/actions3/AMActionRunner3.cpp

```diff
@@ -1045,4 +1085,30 @@ bool AMActionRunner3::runActionImmediatelyInQueue(AMAction3 *action)
 	return true;
 }
 
+bool AMActionRunner3::isActionRunnerPausable() const
+{
+	bool isPausable = isActionRunnerPauseEnabled_;
+
+	if (currentAction_) {
+		AMAction3::State currentActionState = currentAction_->state();
+
+		isPausable = isPausable && currentAction_->canPause() && (currentActionState == AMAction3::Running || currentActionState == AMAction3::Paused);
```

Comment:
The case where `currentActionState == AMAction3::Paused` meant to catch if the action was paused programmatically rather than through the action runner?